### PR TITLE
Don't add property symbol for CLIEvent member.

### LINKED ISSUE #16640: `mfv.LogicalName` throws exceptions for `CLIEvent` properties
Consider the following F# snippet:

```fsharp
type T() = 
    [<CLIEvent>]
    member this.Event = Event<int>().Publish
```

Steps to reproduce:
* get all symbols via checkResults.GetAllUsesOfAllSymbolsInFile()
* iterate via all symbols and access `LogicalName` property

Expected result: no exceptions
Actual result: exception on accessor members:
```
System.Exception: Invalid internal property name: 'remove_Event'
   at Microsoft.FSharp.Core.PrintfModule.PrintFormatToStringThenFail@1448.Invoke(String message)
```

This is a recent regression and it hasn't been happening with FCS `main` from ~November.

### PR DESCRIPTION
**Pull request by nojaf**
## Description

Fixes #16640

The short version here is that an event member will be desugared into multiple members (`add_XYZ`, `remove_XYZ`, ...) and those should not be considered for producing a property symbol.

## Checklist

- [x] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [x] Release notes entry updated:
    > Please make sure to add an entry with short succinct description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/.FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/releae-notes/.Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/.FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Information about the release notes entries format can be found in the [documentation](https://fsharp.github.io/fsharp-compiler-docs/release-notes/About.html).
    > Example:
    > * More inlines for Result module. ([PR #16106](https://github.com/dotnet/fsharp/pull/16106))
    > * Correctly handle assembly imports with public key token of 0 length. ([Issue #16359](https://github.com/dotnet/fsharp/issues/16359), [PR #16363](https://github.com/dotnet/fsharp/pull/16363))
    > *`while!` ([Language suggestion #1038](https://github.com/fsharp/fslang-suggestions/issues/1038), [PR #14238](https://github.com/dotnet/fsharp/pull/14238))

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

File: tests/service/Symbols.fs [hunk 1]
```diff
@@ -1203,3 +1203,24 @@ type T() =
...
+
+        let hasPropertySymbols =
+            allSymbols
+            |> List.exists (fun su ->
+                match su.Symbol with
+                | :? FSharpMemberOrFunctionOrValue as mfv -> mfv.IsProperty
+                | _ -> false
+            )
+
+        Assert.False hasPropertySymbols
```
Comment (auduchinok) [inline tests/service/Symbols.fs:1226]: Hm, does it actually fix the issue? The property is probably expected there (and it should return correct `EventForFSharpProperty` symbol, I think). The unexpected is the other symbol, where the exception in `LogicalName` happens now.

Comment (nojaf) [reply tests/service/Symbols.fs:1226]: Nope, that is not the case. Your unexpected problem occurred for `this.Event`.
And it should not have existed in the first place, which is what the test covers.

Comment (nojaf) [reply tests/service/Symbols.fs:1226]: My original test was:
```
module Meh =
    [<Test>]
    let foo () =
        let _, checkResults = getParseAndCheckResults """
type T() = 
    [<CLIEvent>]
    member this.Event = Event<int>().Publish
"""
        let allSymbols = checkResults.GetAllUsesOfAllSymbolsInFile() |> Seq.toArray
        for su in allSymbols do
            match su.Symbol with
            | :? FSharpMemberOrFunctionOrValue as mfv ->
                printfn "%A" mfv.LogicalName
            | _ -> ()
```

which passed with the same code change in `CheckDeclarations.fs`.
Or course this doesn't really tell much and so I changed it to tailor to the actual problem.

Comment (auduchinok) [reply tests/service/Symbols.fs:1226]: Does this work without type checking each symbol and calling `symbol.LogicalName`?

Comment (nojaf) [reply tests/service/Symbols.fs:1226]: Not sure what you mean here:
image

Comment (auduchinok) [reply tests/service/Symbols.fs:1226]: Ah, OK, so it's not on the base symbol. I thought it was, thanks for checking!

Comment (auduchinok) [reply tests/service/Symbols.fs:1226]: Thanks for clarifying @nojaf. Is there a way to get a single symbol generated for the event?

Comment (nojaf) [reply tests/service/Symbols.fs:1226]: `checkResults.GetSymbolUseAtLocation` would work.

**Review by T-Gro — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]
Thanks!

**Pull request merged**